src/editor/EditorManager.js

```diff
@@ -329,9 +338,11 @@ define(function (require, exports, module) {
      */
     function _insertProviderSorted(array, provider, priority) {
         var index,
+            queryProvider = _addQueryProvider(provider),
             prioritizedProvider = {
                 priority: priority,
-                provider: provider
+                provider: provider,
+                queryProvider: queryProvider
```

Comment:
You shouldn't need to do this, since you already added the function to each provider on line 341 above.  Later you can just loop through the providers and use `provider.queryProvider()`